tests/sagemaker-tests/test_sagemaker_args.py

```diff
@@ -86,3 +90,144 @@ def test_parse_complex_string_with_quotes():
+    test_args_str = [test_cmd] + [str(item) for pair in test_args.items() for item in pair]
+
+    with mock.patch.object(sys, 'argv', test_args_str):
+        arg_parser = get_realtime_infer_parser()
+        # parser will exit with non-zero code
+        with pytest.raises(SystemExit) as excinfo:
+            arg_parser.parse_args()
+        assert excinfo.value.code != 0
+
+def test_validate_realtime_infer_arguments():
+    """ Test the argument logics defined in the validate_realtime_infer_arguments function
+    """
+    test_cmd = "test_sagemaker_args.py"
+    default_args = {'--instance-type': 'ml.c6i.xlarge',
+                    '--instance-count': 1,
+                    '--model-name': 'GSF-Model4Realtime'}
+    required_args = {'--image-url': 'image_url_val',
+                     '--role': 'role_val',
+                     '--region': 'region_val'}
+
+    # Test case 1: normal cases
+    #       1.1: given model_tarfile_s3 and entrypoint-file-name arguments
+    other_args = {'--model-tarfile-s3': 's3://test_bucket/test.tar.gz',
+                  '--entrypoint-file-name': 'entrypoint.py'}
+    test_args = {**required_args, **default_args, **other_args}
+    test_args_str = [test_cmd] + [str(item) for pair in test_args.items() for item in pair]
+
+    with mock.patch.object(sys, 'argv', test_args_str):
+        arg_parser = get_realtime_infer_parser()
+        args = arg_parser.parse_args()
+
+        validate_realtime_infer_arguments(args)
+
+        args_w_vals = {k: v for k, v in vars(args).items() if v is not None}
+        expected_vals = {k.replace('-', '_')[2: ]: v for k, v in test_args.items()}
+        assert args_w_vals == expected_vals
+
+    #       1.2: not provide model_tarfile_s3, but give the 5 arguments
+    other_args = {'--restore-model-path': 'model_path',
+                  '--model-yaml-config-file': 'yaml_config_file',
+                  '--graph-json-config-file': 'json_config_file',
+                  '--upload-tarfile-s3': 'tarfile_s3',
+                  '--infer-task-type': 'node_classification',
+                  '--model-name': 'test-model',     # here overwritten the default, and no '_'
+                  '--model-tarfile-s3': 's3://test_bucket/test.tar.gz',
+                  '--entrypoint-file-name': 'entrypoint.py'}
+    test_args = {**required_args, **default_args, **other_args}
+    test_args_str = [test_cmd] + [str(item) for pair in test_args.items() for item in pair]
+
+    with mock.patch.object(sys, 'argv', test_args_str):
+        arg_parser = get_realtime_infer_parser()
+        args = arg_parser.parse_args()
+
+        validate_realtime_infer_arguments(args)
+
+        args_w_vals = {k: v for k, v in vars(args).items() if v is not None}
+        expected_vals = {k.replace('-', '_')[2: ]: v for k, v in test_args.items()}
+        assert args_w_vals == expected_vals
+
+    # Test case 2: abnormal case. Not provide model-tarfile-s3, and missing one of the 5 arguments
```

Comment:
There is another case, provide `model-tarfile-s3`, but the entrypoint name is not provided.

Reply:
Added.